src/Ghcid.hs

```diff
@@ -307,7 +331,10 @@ runGhcid session waiter termSize termOutput opts@Options{..} = do
                 -- sort error messages by modtime, so newer edits cause the errors to float to the top - see #153
                 errTimes <- sequence [(x,) <$> getModTime x | x <- nubOrd $ map loadFile msgError]
                 let f x = lookup (loadFile x) errTimes
-                return $ sortOn (Down . f) msgError ++ msgWarn
+                return $
+                  if reversed
+                  then msgWarn ++ sortOn f msgError
+                  else sortOn (Down . f) msgError ++ msgWarn
```

Comment:
This doesn't reverse the warnings. I'd rather `(if reversed then reverse else id)` - that way they'll never go out of sync.

Reply:
My understanding is that this `sortOn` only sorts at the module level. All the messages in a module will have the same timestamp. We still rely on GHC to reverse both errors and warnings within a given module. But maybe I understood this bit wrong?

Reply:
True - stable sort. I guess I'd still just, for simplicity have reversed just call reverse, and then it's simple and guaranteed correct. Or are you hoping reverse doesn't reverse in some situations?